travis.sh

```diff
@@ -21,6 +21,16 @@ echo "Testing branch '$TRAVIS_BRANCH' of '$REPOSITORY_NAME' on ROS '$ROS_DISTRO'
 # Helper functions
 source ${CI_SOURCE_PATH}/$CI_PARENT_DIR/util.sh
 
+# Split for different tests
+for t in $TEST; do
+    case "$t" in
+        "clang-format")
+            sudo bash ${CI_SOURCE_PATH}/$CI_PARENT_DIR/check_clang_format.sh || exit 1
+            exit 0 # This runs as an independent job, do not run regular travis test
+        ;;
+    esac
+done
```

Comment:
by skipping docker you run the risk that the ubuntu version that travis is using could be out of date (it was 12.04 for a very long time) and that the exact command isn't available on certain OSes:

> apt-get -qq install -y clang-format-3.8

I'm not convinced loading a docker container for this is that big a cost... they are cached on Travis anyway and pre-built by dockerhub

Reply:
It's a minute extra or so. Sure, not that big a cost.

Reply:
Also, you don't need to specify `ROS_DISTRO` and `ROS_REPO` environment variables needed to run the docker container.

Reply:
Nevertheless, I reverted this part.